crates/red_knot_python_semantic/resources/mdtest/type_qualifiers/classvar.md

```diff
@@ -0,0 +1,93 @@
+# `typing.ClassVar`
+
+[`typing.ClassVar`] is a type qualifier that is used to indicate that a class variable may not be
+written to from instances of that class.
+
+This test makes sure that we discover the type qualifier while inferring types from an annotation.
+For more details on the semantics of pure class variables, see [this test](../attributes.md).
+
+## Basic
+
+```py
+from typing import ClassVar, Annotated
+
+class C:
+    a: ClassVar[int] = 1
+    b: Annotated[ClassVar[int], "the annotation for b"] = 1
+    c: ClassVar[Annotated[int, "the annotation for c"]] = 1
+    d: ClassVar = 1
+    e: "ClassVar[int]" = 1
+
+reveal_type(C.a)  # revealed: int
+reveal_type(C.b)  # revealed: int
+reveal_type(C.c)  # revealed: int
+# TODO: should be Unknown | Literal[1]
+reveal_type(C.d)  # revealed: Unknown
```

Comment:
@carljm I haven't tackled this TODO yet. I think it requires changes that are rather orthogonal to what I did here, but let me know if you think it should be solved in this PR.